File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/inbound/zeebe/bike/CheckBikeAvailabilityWorker.java
package io.miragon.example.adapter.inbound.zeebe.bike;

import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi;
import io.miragon.example.application.port.inbound.bike.CheckBikeAvailabilityUseCase;
import io.miragon.example.domain.bike.BikeSubscriptionId;
import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.Map;
import java.util.UUID;

@Component
public class CheckBikeAvailabilityWorker {

    private static final Logger log = LoggerFactory.getLogger(CheckBikeAvailabilityWorker.class);

    private final CheckBikeAvailabilityUseCase useCase;

    public CheckBikeAvailabilityWorker(CheckBikeAvailabilityUseCase useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = BikeSubscriptionSignupProcessApi.TaskTypes.BIKE_CHECK_AVAILABILITY)
    public Map<String, Object> handle(@Variable String subscriptionId) {
        log.info("Checking bike availability for subscription: {}", subscriptionId);
        boolean available = useCase.checkAvailability(new BikeSubscriptionId(UUID.fromString(subscriptionId)));
        return Map.of("bikeAvailable", available);
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/inbound/zeebe/bike/CheckBikeAvailabilityWorker.kt
package io.miragon.example.adapter.inbound.zeebe.bike

import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi
import io.miragon.example.application.port.inbound.bike.CheckBikeAvailabilityUseCase
import io.miragon.example.domain.bike.BikeSubscriptionId
import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class CheckBikeAvailabilityWorker(
    private val useCase: CheckBikeAvailabilityUseCase
) {
    private val log = KotlinLogging.logger {}

    @JobWorker(type = BikeSubscriptionSignupProcessApi.ServiceTasks.BIKE_CHECK_AVAILABILITY)
    fun handle(@Variable subscriptionId: String): Map<String, Any> {
        log.info { "Checking bike availability for subscription: $subscriptionId" }
        val available = useCase.checkAvailability(BikeSubscriptionId(UUID.fromString(subscriptionId)))
        return mapOf("bikeAvailable" to available)
    }
}
